### DISCUSSION TITLE
Translations for Formly options labels

### DISCUSSION THREAD

**Discussion by uginy** [ngx-formly/ngx-formly#2772] [General]
Hi!

Howto?
1. translate options' labels initially (when page refresh or enter to the page)
2. translate options' labels dynamically when i change language with i18n (ngx-translate module)?

My current solution is not pretty good:

{
   className: 'p-field p-col',
   key: 'gender',
   type: 'select',
   templateOptions: {
       required: true,
       label: 'PAGES.persons.fields.gender',
       options: [
             {
                label: this.translate.instant('PAGES.persons.gender.select'),
                value: ''
             },
              {
                label: this.translate.instant('PAGES.persons.sex.male'),
                value: 'male'
             },
             {
                 label: this.translate.instant('PAGES.persons.sex.female'),
                 value: 'female'
             },
      ];
   },
   hooks: {
        onChanges: (field: FormlyFieldConfig) => {
            this.translate.onLangChange.subscribe(lang => {
            field.templateOptions.options = [
                                                            {
                label: this.translate.instant('PAGES.persons.gender.select'),
                value: ''
             },
              {
                label: this.translate.instant('PAGES.persons.sex.male'),
                value: 'male'
             },
             {
                 label: this.translate.instant('PAGES.persons.sex.female'),
                 value: 'female'
             },
           ];
         }
        );
      }
    }
}

Could you advice how to make options labels translated properly if i change language with ngx-translate?

**Comment by aitboudad** [comment]
Better rely on `stream` instead of `instant` method you may check our solutions for i18n use two approaches:

- https://formly.dev/examples/advanced/i18n
- https://formly.dev/examples/advanced/i18n-alternative
